Automattic/wp-calypso
# Ensure Woo Express 1USD introductory offers don't include bundle domain feature

## Proposed Changes

We are planning to offer introductory offers for Woo Express plans, and we won't include a free domain, as it would cost us too much upfront, and it's not very useful to get three months after starting your plan subscription. As a result, we don't want to remove the "Free domain for one year" feature in the plans grid when we know that the user is being shown an introductory offer for a Woo Express annual plan.

* Given the above, this PR removes the custom domain feature in the Woo Express plans grid when we are showing an introductory offer for an annual Woo Express plan

**Note:** While testing this change, I discovered a separate bug where we're not showing the correct pricing for Woo Express plans that were purchased using the introductory offer. The issue there is related to a subtle back-end bug that results in the introductory offer not being returned correctly, as well as some related display issues for the current plan on the site.
  - I am working on the API fix in D123944-code, and will create a dedicated PR for the UI changes

## Testing Instructions

* Ensure that you have a Woo Express trial site that has never purchased a Woo Express plan before (i.e. is eligible for the introductory offer) -- if you don't have one, you can create one by visiting (https://wordpress.com/setup/wooexpress)
* Run this branch via Calypso.live per the instructions in [this comment](https://github.com/Automattic/wp-calypso/pull/81506#issuecomment-1709971707), or run the branch locally
* Navigate to _Upgrades_ -> _Plans_ for your Woo Express trial site while logged in with your Automattic employee account (so you can see the introductory offers)
* You should see the monthly plans shown by default
* Verify that you're seeing the introductory offer pricing for the first three months, and that there is no mention of a free domain
* Click on the toggle to switch to yearly plans
* Verify that there is no mention of a free domain for either plan

## Pre-merge Checklist



- [N/A] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [N/A] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?
